compiler/noirc_frontend/src/elaborator/expressions.rs

```diff
@@ -669,124 +683,119 @@ impl Elaborator<'_> {
         let method_name_location = method_call.method_name.location();
         let method_name = method_call.method_name.as_str();
         let check_self_param = true;
-        match self.lookup_method(
+
+        let method_ref = self.lookup_method(
```

Comment:
This and the rest of the changes in this method were done just to reduce nesting